emigration.lua

```diff
@@ -36,16 +37,91 @@ function desert(u,method,civ)
         line = line.."joined the merchants"
         u.flags1.merchant = true
         u.civ_id = civ
-    elseif method == 'diplomat' then
-        line = line.."followed the diplomat"
-        u.flags1.diplomat = true
-        u.civ_id = civ
     else
         line = line.."abandoned the settlement in search of a better life."
-        u.civ_id = -1
+        u.civ_id = civ
         u.flags1.forest = true
+        u.flags2.visitor = true
         u.animal.leave_countdown = 2
     end
+
+    local hf_id = u.hist_figure_id
+    local hf = df.historical_figure.find(u.hist_figure_id)
+    local fort_ent = df.global.ui.main.fortress_entity
+    local civ_ent = df.historical_entity.find(hf.civ_id)
+
+    local newent_id = -1
+    local newsite_id = -1
+
+    -- free owned rooms
+    for i = #u.owned_buildings-1, 0, -1 do
+        local temp_bld = df.building.find(u.owned_buildings[i].id)
+        dfhack.buildings.setOwner(temp_bld, nil)
```

Comment:
side note: I got the `dfhack.buildings.setOwner` function working for v50 just last week. good timing.